massa-execution-worker/src/execution.rs

```diff
@@ -1260,6 +1381,13 @@ impl ExecutionState {
             self.mip_store.clone(),
         );
 
+        // Deferred calls
+        let calls = execution_context.deferred_calls_advance_slot(
+            slot.clone(),
+            DEFERRED_CALL_MAX_FUTURE_SLOTS,
+            self.config.thread_count,
+        );
+
         // Get asynchronous messages to execute
         let messages = execution_context.take_async_batch(
```

Comment:
I think you need to update the take_async_batch here, because for now we only take config.max_async_gas worth of messages

Reply:
added ```rust self.config.max_async_gas.saturating_sub(calls.slot_gas)```